pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -926,9 +927,13 @@ public void unloadNamespaceBundlesGracefully() {
     }
 
     public void unloadNamespaceBundlesGracefully(int maxConcurrentUnload, boolean closeWithoutWaitingClientDisconnect) {
+        if (unloaded) {
+            return;
+        }
```

Comment:
is there a need to prevent multiple calls to concurrently call this method? This solution doesn't seem very effective.

Reply:
It's not in a critical path. Here I use a volatile field just because I realized shutdown could also be triggered via an admin API (see `BrokersBase#shutDownBrokerGracefully`). Therefore, to avoid thread-safety issues, I used volatile here.